site/docs/main/config/rate-limiting.md

```diff
@@ -128,3 +148,118 @@ spec:
         - name: x-contour-ratelimited
           value: "true"
 ```
+
+## Global Rate Limiting
+
+The `HTTPProxy` API also supports defining global rate limit policies on routes and virtual hosts.
+
+In order to use global rate limiting, you must first select and deploy an external rate limit service (RLS).
+There is an [Envoy rate limit service implementation][2], but any service that implements the [RateLimitService gRPC interface][3] is supported.
+
+### Configuring an exernal RLS with Contour
+
+Once you have deployed your RLS, you must configure it with Contour.
+
+Define an extension service for it (substituting values as appropriate):
+```yaml
+apiVersion: projectcontour.io/v1alpha1
+kind: ExtensionService
+metadata:
+  namespace: projectcontour
+  name: ratelimit
+spec:
+  protocol: h2
+  services:
+    - name: ratelimit
+      port: 8081
+```
+
+Now add a reference to it in the Contour config file:
+```yaml
+rateLimitService:
+  extensionService: projectcontour/ratelimit
+  domain: contour
```

Comment:
Maybe describe these fields a bit? Why would I change the `domain`? What does `failOpen` mean, etc.

Reply:
Just noticed it's defined later on, but maybe redefine or link?

Reply:
Added comments.